generators/python/core_utilities/shared/http_client.py

```diff
@@ -242,6 +242,11 @@ def request(
         if (request_files is None or len(request_files) == 0) and force_multipart:
             request_files = FORCE_MULTIPART
 
+        # Filter None values from data body for multipart/form requests to prevent
+        # httpx from converting None to empty strings
+        if data_body is not None and isinstance(data_body, typing.Mapping) and (request_files or force_multipart):
```

Comment:
let's refactor this to be a helper function in this file (we call the same thing 4 times